stress/shell/src/main/java/alluxio/stress/cli/suite/AbstractMaxThroughput.java

```diff
@@ -11,35 +11,110 @@
 
 package alluxio.stress.cli.suite;
 
+import alluxio.ClientContext;
+import alluxio.client.job.JobMasterClient;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.stress.Parameters;
 import alluxio.stress.TaskResult;
+import alluxio.stress.cli.Benchmark;
 import alluxio.stress.common.AbstractMaxThroughputSummary;
 import alluxio.stress.common.GeneralBenchSummary;
-import alluxio.stress.common.GeneralParameters;
+import alluxio.util.CommonUtils;
+import alluxio.util.ConfigurationUtils;
+import alluxio.worker.job.JobMasterClientContext;
 
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.ParametersDelegate;
+import com.google.common.collect.Lists;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 /**
- * abstract class for MaxThroughput stressBench.
+ * abstract class for MaxThroughput benchmark.
+ *
  * @param <T> the MaxThroughput bench result summary
  * @param <S> the general Bench Summary with common method
  * @param <P> the stress bench parameter
  * @param <Q> the single task result
  */
 public abstract class AbstractMaxThroughput<Q extends TaskResult, T extends
-    AbstractMaxThroughputSummary<P, S>, S extends GeneralBenchSummary<Q>, P extends
-    GeneralParameters> extends Suite<T> {
+    AbstractMaxThroughputSummary<P, S>, S extends GeneralBenchSummary<Q>,
+    P extends Parameters> extends Benchmark<T> {
   protected static final Logger LOG = LoggerFactory.getLogger(AbstractMaxThroughput.class);
 
-  protected int getBestThroughput(int initialThroughput, T summary, List<String> baseArgs,
-      int numWorkers) throws Exception {
+  @ParametersDelegate
+  protected P mParameters;
+
+  protected T mMaxThroughputResult;
+
+  protected int mTargetThroughput = 1000;
```

Comment:
we may want to discuss how we can set the initial targetThroughput here.  Probably put in another class like MaxThroughput parameter?